dirsrvtests/tests/suites/setup_ds/dscreate_test.py

```diff
@@ -115,3 +127,133 @@ def test_setup_ds_minimal(topology):
     topology.standalone.start()
     # Okay, actually remove the instance
     remove_ds_instance(topology.standalone)
+
+def write_file(fname, pwd, is_runnable, lines):
+    log.debug(f'Creating file {fname} with:')
+    with open(fname, 'wt') as f:
+        for line in lines:
+            log.debug(line)
+            f.write(line+'\n')
+        os.chown(fname, pwd.pw_uid, pwd.pw_gid)
+        if (is_runnable):
+            os.chmod(fname, 0o755)
+    log.debug(f'End of file: {fname}')
+
+def test_setup_ds_as_non_root():
+    """Test creating an instance as a non root user
+
+    :id: c727998e-a960-11ec-898e-482ae39447e5
+    :setup: no instance
+    :steps:
+        1. Add linux user NON_ROOT_USER if it does not already exist
+        2. Create a temporary directory
+        3. Create test.sh script that
+             Set Paths
+             Run dscreate prepare-subtree
+             Run dscreate from-file
+             Add a backend
+             Search users in backend and store output in a file
+             Stop the instance
+        4. Create a dscreate template file
+        5. Run su - NON_ROOT_USER test.sh
+        6. Check that pid file exists and kill the associated process
+        7. Check demo_user is in the search result
+        8. Check that test.sh returned 0
+        9. (Implicit task): remove the temporary
+
+
+    :expectedresults:
+        1. No error.
+        2. No error.
+        3. No error.
+        4. No error.
+        5. No error.
+        6. Should fail to kill the process (That is supposed to be stopped)
+        7. demo_user should be in search result
+        8. return code should be 0
+        9. No error.
+
+    """
+
+    # Add linux user NON_ROOT_USER if it does not already exist
+    NON_ROOT_USER='user1'
+    try:
+        pwd_nru = pwd.getpwnam(NON_ROOT_USER)
+    except KeyError:
+        subprocess.run(('/usr/sbin/useradd', NON_ROOT_USER))
+        pwd_nru = pwd.getpwnam(NON_ROOT_USER)
+
+    # Create a temporary directory
+    with TemporaryDirectory() as dir:
+        os.chown(dir, pwd_nru.pw_uid, pwd_nru.pw_gid)
+        # Prepare test script
+        if DEBUGGING:
+            dbg_script = (
+                'savedir()',
+                '{',
+                f'    cp -R {dir} /tmp/dbg_test_setup_ds_as_non_root',
+                '}',
+                'trap savedir exit',
+                'trap savedir err',
+            )
+        else:
+            dbg_script = ( )
+        # Export current path and python path in test script
+        # to insure that right binary/libraries are used
+        path = os.environ['PATH']
+        pythonpath = os.environ['PYTHONPATH']
```

Comment:
Should use os.getenv('PYTHONPATH', '') to handle the case where PYTHPATH is not defined.